frontend/packages/kubevirt-plugin/src/utils/sort.ts

```diff
@@ -46,3 +47,77 @@ export const ignoreCaseSort = <T>(
   };
   return array.sort((a, b) => resolve(a).localeCompare(resolve(b)));
 };
+
+const getOSVersion = (osName: string): number[] =>
+  osName
+    .split(/\D/)
+    .filter((x) => x)
+    .map((num) => parseInt(num, 10));
+
+/**
+ *
+ *
+ * Compare the numbers between the two versions by the order of their appearance
+ * in the OS name.
+ *
+ * For example:
+ * osVersion1: [10,2] for OS: 'Windows 10 R2',
+ * osVersion2: [10] for OS: 'Windows 10',
+ * (return 1)
+ *
+ * osVersion1: [9,10] for OS: 'ubuntu9.10',
+ * osVersion2: [10,4] for OS: 'ubuntu10.04',
+ * (return -1)
+ *
+ * return 0 when equal.
+ *
+ */
+export const compareVersions = (osVersion1: number[], osVersion2: number[]): number => {
+  if (!osVersion1 && !osVersion2) {
+    return 0;
+  }
+
+  if (!osVersion1) {
```

Comment:
these two ifs should just fill osVersion with empty array - you don't know what will come in the other version. 

It could be [0] or [null]

can you please add a test for that?